azurerm/internal/services/network/express_route_circuit_connection_resource_test.go

```diff
@@ -0,0 +1,245 @@
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_circuit_connection" "import" {
+  name                = azurerm_express_route_circuit_connection.test.name
+  peering_id          = azurerm_express_route_circuit_connection.test.peering_id
+  peer_peering_id     = azurerm_express_route_circuit_connection.test.peer_peering_id
+  address_prefix_ipv4 = azurerm_express_route_circuit_connection.test.address_prefix_ipv4
+}
+`, r.basic(data))
+}
+
+func (r ExpressRouteCircuitConnectionResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_circuit_connection" "test" {
+  name                = "acctest-ExpressRouteCircuitConn-%d"
+  peering_id          = azurerm_express_route_circuit_peering.test.id
+  peer_peering_id     = azurerm_express_route_circuit_peering.peer_test.id
+  address_prefix_ipv4 = "192.169.9.0/29"
+  authorization_key   = "846a1918-b7a2-4917-b43c-8c4cdaee006a"
+  address_prefix_ipv6 = "aa:bb::/125"
+}
+`, r.template(), data.RandomInteger)
+}
+
+func (r ExpressRouteCircuitConnectionResource) updateAddressPrefixIPv6(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_circuit_connection" "test" {
+  name                = "acctest-ExpressRouteCircuitConn-%d"
+  peering_id          = azurerm_express_route_circuit_peering.test.id
+  peer_peering_id     = azurerm_express_route_circuit_peering.peer_test.id
+  address_prefix_ipv4 = "192.169.9.0/29"
+  authorization_key   = "846a1918-b7a2-4917-b43c-8c4cdaee006a"
+  address_prefix_ipv6 = "aa:cc::/125"
+}
+`, r.template(), data.RandomInteger)
+}
+
+func (r ExpressRouteCircuitConnectionResource) template() string {
+	rg := os.Getenv("ARM_TEST_DATA_RESOURCE_GROUP")
+	circuitName1 := os.Getenv("ARM_TEST_CIRCUIT_NAME_FIRST")
+	circuitName2 := os.Getenv("ARM_TEST_CIRCUIT_NAME_SECOND")
+
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_express_route_circuit_peering" "test" {
+  peering_type                  = "AzurePrivatePeering"
+  express_route_circuit_name    = "%[1]s"
+  resource_group_name           = "%[2]s"
+  shared_key                    = "ItsASecret"
+  peer_asn                      = 100
+  primary_peer_address_prefix   = "192.168.1.0/30"
+  secondary_peer_address_prefix = "192.168.1.0/30"
+  vlan_id                       = 100
+}
+
+resource "azurerm_express_route_circuit_peering" "peer_test" {
+  peering_type                  = "AzurePrivatePeering"
+  express_route_circuit_name    = "%[3]s"
+  resource_group_name           = "%[2]s"
+  shared_key                    = "ItsASecret"
+  peer_asn                      = 100
+  primary_peer_address_prefix   = "192.168.1.0/30"
+  secondary_peer_address_prefix = "192.168.1.0/30"
+  vlan_id                       = 100
+}
+`, circuitName1, rg, circuitName2)
+}
+
+func skipExpressRouteCircuitConnection() bool {
```

Comment:
Can we pass in the `t` and call t.Skip() inside this function rather than checking outside?

Reply:
Updated